### PR TITLE
rlm_lua: Moving to single table 'fr.{}'

### PR DESCRIPTION
**Pull request by jpereira**
Now we have the tables.

fr.log.{ debug(), info(), warn(), error() }
fr.request.{} Copy of the current REQUEST
fr.rcode.{} Constants with all return codes.

### CODE REVIEW

**Comment by arr2036** [comment]
Can you rebase this commit series and fix the build failures

**Comment by arr2036** [comment]
Could you rebase again.  The build should pass now unless there's an issue with the new code.

**Comment by jpereira** [comment]
@arr2036 done. btw, the current `fr.*` table has those fields/functions as can be seen here.

```
#mod8.lua: init
# fr.{}
type(fr) = table
for k,v in pairs(fr)
	log	table: 0x405aac90
	rcode	table: 0x405aada0
	request	table: 0x405aaef8

# fr.rcode.{}
type(fr.rcode) = table
	fr.rcode.noop       = 7
	fr.rcode.handled    = 3
	fr.rcode.ok         = 2
	fr.rcode.reject     = 0
	fr.rcode.fail       = 1
	fr.rcode.invalid    = 4
	fr.rcode.userlock   = 5
	fr.rcode.notfound   = 6
	fr.rcode.updated    = 8

# fr.log.{}
type(fr.log) = table
for k,v in pairs(fr.log)

Mon Mar 18 17:56:37 2019 : Debug : Powered by Luajit+FFI & fr_log()
Mon Mar 18 17:56:37 2019 : Warn  : You can't modify the table 'fr.log.$func()' (read-only)
Mon Mar 18 17:56:37 2019 : Debug : Powered by Luajit+FFI & fr_log()
# fr.request.{}
type(fr.request) = table
for k,v in fr.request.pairs()
	User-Name	caipirinha
	User-Password	gelo
	Called-Station-Id	aa:bb:cc:dd:ee:ff
	Framed-IPv6-Prefix	11:22:33:44:55:66:77:88/128

#mod8.lua: returning fr.rcode.noop
```

**Comment by jpereira** [comment]
@arr2036 done. btw, the current `fr.*` table has those fields/functions as can be seen here.

```
#mod8.lua: init
# fr.{}
type(fr) = table
for k,v in pairs(fr)
	log	table: 0x405aac90
	rcode	table: 0x405aada0
	request	table: 0x405aaef8

# fr.rcode.{}
type(fr.rcode) = table
	fr.rcode.noop       = 7
	fr.rcode.handled    = 3
	fr.rcode.ok         = 2
	fr.rcode.reject     = 0
	fr.rcode.fail       = 1
	fr.rcode.invalid    = 4
	fr.rcode.userlock   = 5
	fr.rcode.notfound   = 6
	fr.rcode.updated    = 8

# fr.log.{}
type(fr.log) = table
for k,v in pairs(fr.log)

Mon Mar 18 17:56:37 2019 : Debug : Powered by Luajit+FFI & fr_log()
Mon Mar 18 17:56:37 2019 : Warn  : You can't modify the table 'fr.log.$func()' (read-only)
Mon Mar 18 17:56:37 2019 : Debug : Powered by Luajit+FFI & fr_log()
# fr.request.{}
type(fr.request) = table
for k,v in fr.request.pairs()
	User-Name	caipirinha
	User-Password	gelo
	Called-Station-Id	aa:bb:cc:dd:ee:ff
	Framed-IPv6-Prefix	11:22:33:44:55:66:77:88/128

#mod8.lua: returning fr.rcode.noop
```

File: src/tests/modules/lua/mod1.lua [hunk 1]
```diff
@@ -1,8 +1,8 @@
 function authorize()
 	if not type(fr) == "table" then
 		print("error: the 'fr' should be a table")
-		return fr.fail
+		return "fail"
 	end
 
-	return fr.ok
+	return "noop"
```
Comment (alandekok) [inline src/tests/modules/lua/mod1.lua]: tests that pass should be `ok` not `noop`

Comment (jpereira) [reply src/tests/modules/lua/mod1.lua]: fixed.

File: src/modules/rlm_lua/aux.c [hunk 2]
```diff
@@ -166,33 +180,32 @@ int fr_lua_aux_jit_log_register(rlm_lua_t const *inst, lua_State *L)
...
-			} fr_log_type_t;\
-			int fr_log(fr_log_type_t lvl, char const *fmt, ...);\
-			]]\
-		fr_srv = ffi.load(\"freeradius-server\")\
+			int fr_lua_aux_wrapper_fr_log(int type, char const *msg); \
+		]]\
 		fr_lua = ffi.load(\"freeradius-lua\")\
-		fr.log = {}\
-		fr.log.debug = function(msg)\
-		   fr_srv.fr_log(16, \"%s\", msg)\
```
Comment (arr2036) [inline src/modules/rlm_lua/aux.c]: Ah, so it was the variadic arguments that caused the crash?

**Pull request merged**